### PR TITLE
fix: iframe to component template

### LINKED ISSUE #2000: [Bug]: build error after running iframe-to-component schematic
### Package name

third-party

### Package version

11.0.0

### Reproduction steps

```
npm create @o3r test-3rd-party
cd test-3rd-party/
npm run ng g app webapp
npm run ng g component comp
npm run ng add @o3r/third-party --project-name=webapp
npm run ng g iframe-to-component -- --path=apps/webapp/src/components/comp/comp.component.ts
npm run build -w webapp
```

### Current result

```
> webapp@0.0.0-placeholder build
> ng build

Application bundle generation failed. [3.525 seconds]

✘ [ERROR] TS2564: Property 'bridge' has no initializer and is not definitely assigned in the constructor. [plugin angular-compiler]

    apps/webapp/src/components/gigi/gigi.component.ts:18:12:
      18 │     private bridge: IframeBridge;
         ╵             ~~~~~~


✘ [ERROR] TS2304: Cannot find name 'ViewChild'. [plugin angular-compiler]

    apps/webapp/src/components/gigi/gigi.component.ts:19:5:
      19 │     @ViewChild('frame')
         ╵      ~~~~~~~~~


✘ [ERROR] TS2552: Cannot find name 'ElementRef'. Did you mean 'Element'? [plugin angular-compiler]

    apps/webapp/src/components/gigi/gigi.component.ts:20:19:
      20 │     private frame: ElementRef<HTMLIFrameElement>;
         ╵                    ~~~~~~~~~~

  'Element' is declared here.

    node_modules/typescript/lib/lib.dom.d.ts:7894:12:
      7894 │ declare var Element: {
           ╵             ~~~~~~~


```

### Expected result

 `Application bundle generation complete. [2.156 seconds]`

### Additional comments

There are some issues in the schematic templates.

### PR DESCRIPTION
**Pull request by vscaiceanu-1a**
## Related issues

- :bug: Fixes #2000

### CODE REVIEW

**Review by kpanot — APPROVED** [review]

File: packages/@o3r/third-party/testing/jest.config.it.js
Comment (fpaul-1A) [inline packages/@o3r/third-party/testing/jest.config.it.js:1]: `packages/@o3r/third-party/jest.config.js` should be updated as well

File: packages/@o3r/third-party/schematics/index.it.spec.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const relativeApplicationPath = path.relative(workspacePath, applicationPath);
+
+    packageManagerExec({script: 'ng', args: ['add', `@o3r/third-party@${o3rVersion}`,'--project-name', appName, '--skip-confirmation']}, execAppOptions);
+    expect(() => packageManagerInstall(execAppOptions)).not.toThrow();
+
+    const diff = getGitDiff(workspacePath);
+    expect(diff.modified).toContain('package.json');
+    expect(diff.modified).toContain('angular.json');
+    expect(diff.modified).toContain('apps/test-app/package.json');
+    expect(diff.modified.length).toBe(4); // with yarn.lock or package-lock.json
```
Comment (fpaul-1A) [inline packages/@o3r/third-party/schematics/index.it.spec.ts:31]: we can check for the actual file to be created https://github.com/AmadeusITGroup/otter/pull/1995/files#diff-e0410f3a4b89d7e26e783977c7180b91cb6d2b6e9587a36bd252dc4e4157616dR59

File: packages/@o3r/third-party/schematics/iframe-to-component/index.ts [hunk 2]
```diff
@@ -126,8 +128,8 @@ export function ngAddIframeFn(options: NgAddIframeSchematicsSchema): Rule {
 
                   /* eslint-disable indent */
                   const propertiesToAdd = generateClassElementsFromString(`
-                    @ViewChild('frame') private frame: ElementRef<HTMLIFrameElement>;
-                    private bridge: IframeBridge;
+                    @ViewChild('frame') private frame!: ElementRef<HTMLIFrameElement>;
```
Comment (matthieu-crouzet) [inline packages/@o3r/third-party/schematics/iframe-to-component/index.ts:131]: Could we use [viewChild](https://angular.dev/api/core/viewChild#) instead, to not have a dangerous `!`

File: packages/@o3r/third-party/schematics/index.it.spec.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/**
+ * Test environment exported by O3rEnvironment, must be first line of the file
+ * @jest-environment @o3r/test-helpers/jest-environment
+ * @jest-environment-o3r-app-folder test-app-application
```
Comment (fpaul-1A) [inline packages/@o3r/third-party/schematics/index.it.spec.ts:4]:
```suggestion
 * @jest-environment-o3r-app-folder test-app-third-party
```

**Review by fpaul-1A — APPROVED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

File: packages/@o3r/third-party/testing/jest.config.it.js
Comment (fpaul-1A) [reply packages/@o3r/third-party/testing/jest.config.it.js:1]: my bad, it should definitely not be updated

File: packages/@o3r/third-party/testing/jest.config.it.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const { join } = require('node:path');
+const getJestConfig = require('../../../../jest.config.it').getJestConfig;
```
Comment (kpanot) [inline packages/@o3r/third-party/testing/jest.config.it.js:2]:
```suggestion
const { getJestConfig } = require('../../../../jest.config.it');
```

to be consistent with the previous line :D

**Review by kpanot — APPROVED** [review]

File: packages/@o3r/third-party/testing/jest.config.it.js (hunk 4 above)
Comment (vscaiceanu-1a) [reply packages/@o3r/third-party/testing/jest.config.it.js:2]: let's align all packages in another PR :-) 
image

**Pull request merged**